packages/contracts/contracts/L1/Colosseum.sol

```diff
@@ -209,108 +273,145 @@ contract Colosseum is Initializable, Semver {
+                challenge.segments[_pos] == srcOutputRoot,
+                "Colosseum: the source segment must be matched"
+            );
+            require(
+                challenge.segments[_pos + 1] != dstOutputRoot,
+                "Colosseum: the destination segment must not be matched"
+            );
+        }
 
-        bytes32 blockHash = Hashing.hashBlockHeader(_publicInput, _rlps);
         require(
-            _srcOutputRootProof.nextBlockHash == blockHash,
+            _srcOutputRootProof.nextBlockHash == _dstOutputRootProof.blockHash,
             "Colosseum: the block hash must be matched"
         );
 
-        bytes32[] memory dummyHashes;
-        if (_publicInput.txHashes.length < MAX_TXS) {
-            dummyHashes = Hashing.generateDummyHashes(
-                DUMMY_HASH,
-                MAX_TXS - _publicInput.txHashes.length
-            );
-        }
+        // TODO(pangssu): waiting for the new Verifier.sol to complete.
+        // require(ZK_VERIFIER.verify(_proof, _pair), "Colosseum: invalid proof");
 
-        // TODO(chokobole): check transaction root hash
-        // TODO(chokobole): use this public input hash require(_proof[4] == _publicInputHash);
-        bytes32 _publicInputHash = Hashing.hashPublicInput(
-            _srcOutputRootProof.stateRoot,
-            _publicInput,
-            CHAIN_ID,
-            dummyHashes
-        );
+        _validatePublicInput(_srcOutputRootProof, _publicInput, _rlps, _pair[4]);
+        challenge.outputRoot = _outputRoot;
 
-        require(ZK_VERIFIER.verify(_proof, _pair), "Colosseum: invalid proof");
+        // TODO(pangssu): call `createTransaction` to guardian
 
-        L2_ORACLE.deleteL2Outputs(challenge.outputIndex);
-        emit ProofCompleted(challengeId, challenge.outputIndex);
-        _closeChallenge(challengeId, challenge);
+        emit Proven(_outputIndex, _outputRoot);
     }
 
-    function asserterTimeout() external {
-        uint256 challengeId = latestChallengeId;
-        Types.Challenge storage challenge = challenges[challengeId];
-        _validateTurn(challenge, ValidateTurnMode.NORMAL);
-        L2_ORACLE.deleteL2Outputs(challenge.outputIndex);
-        _closeChallenge(challengeId, challenge);
+    /**
+     * @notice Closes the challenge because the challenger timed out.
+     *         The winner is the asserter.
+     *
+     * @param _outputIndex Index of the L2 checkpoint output.
+     */
+    function challengerTimeout(uint256 _outputIndex) external {
+        Types.Challenge storage challenge = challenges[_outputIndex];
+        _validateTurn(challenge);
+        _closeChallenge(_outputIndex, challenge);
     }
 
-    function challengerTimeout(uint256 _challengeId) external {
-        Types.Challenge storage challenge = challenges[_challengeId];
-        _validateTurn(challenge, ValidateTurnMode.NORMAL);
-        _closeChallenge(_challengeId, challenge);
+    /**
+     * @notice Approve the challenge to replace output.
```

Comment:
It's very minor but in order to other comments within this file, it seems to be more appropriate to write it `Approves`.